libevmasm/RuleList.h

```diff
@@ -374,6 +390,32 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	rules.push_back({
+		// AND(A, SHR(B, X)) -> SHR(B, X) if 2^(256-B)-1 == A
+		{Instruction::AND, {A, {Instruction::SHR, {B, X}}}},
+		[=]() -> Pattern { return {Instruction::SHR, {B, X}}; },
+		false,
+		[=]() {
+			if (B.d() > std::numeric_limits<unsigned>::max())
+				return false;
+			unsigned bAsUint = static_cast<unsigned>(B.d());
+			return (A.d() & (u256(-1) >> bAsUint)) == (u256(-1) >> bAsUint);
+		}
+	});
+
+	rules.push_back({
+		// AND(SHR(A, X), B) -> SHR(A, X) if 2^(256-B)-1 == A
+		{Instruction::AND, {{Instruction::SHR, {B, X}}, A}},
+		[=]() -> Pattern { return {Instruction::SHR, {B, X}}; },
+		false,
+		[=]() {
```

Comment:
Since the lambda now is the same you should pull it out and only define it once :-). But I guess that was the plan anyways and is just not done yet.